reco_utils/recommender/deeprec/io/dkn_iterator.py

```diff
@@ -187,6 +204,96 @@ def load_data_from_file(self, infile):
+                    click_news_val_batch,
+                    candidate_news_entity_index_batch,
+                    click_news_entity_index_batch,
+                    impression_id_list
+                )
+                data_size = self.batch_size
+                yield self.gen_feed_dict(res), impression_id_list, data_size
+
+    def load_infer_data_from_file(self, infile):
+        """Read and parse data from a file for infer document embedding.
+
+        Args:
+            infile (str): text input file. Each line in this file is an instance.
+
+        Returns:
+            obj: An iterator that will yields parsed results, in the format of graph feed_dict.
+            List: news id list
+            Int: size of the data in a batch
+        """
+        newsid_list = []
+        candidate_news_index_batch = []
+        candidate_news_val_batch = []
+        candidate_news_entity_index_batch = []
+        cnt = 0
+        with tf.gfile.GFile(infile, "r") as rd:
+            while True:
+                line = rd.readline()
+                if not line:
+                    break
+                newsid, word_index, entity_index = line.strip().split(" ")
+                newsid_list.append(newsid)
+                candidate_news_index = []
+                candidate_news_val = []
+                candidate_news_entity_index = []
+                for item in word_index.split(","):
+                    candidate_news_index.append(int(item))
+                    candidate_news_val.append(float(1))
+                for item in entity_index.split(","):
+                    candidate_news_entity_index.append(int(item))
+
+                candidate_news_index_batch.append(candidate_news_index)
+                candidate_news_val_batch.append(candidate_news_val)
+                candidate_news_entity_index_batch.append(candidate_news_entity_index)
+
+                cnt += 1
+                if cnt >= self.batch_size:
+                    res = self._convert_infer_data(
+                        candidate_news_index_batch,
+                        candidate_news_val_batch,
+                        candidate_news_entity_index_batch,
+                    )
+                    data_size = self.batch_size
+                    yield self.gen_infer_feed_dict(res), newsid_list, data_size
+                    candidate_news_index_batch = []
+                    candidate_news_val_batch = []
+                    candidate_news_entity_index_batch = []
+                    newsid_list = []
+                    cnt = 0
+
+            if cnt > 0:
+                data_size = cnt
+                while cnt < self.batch_size:
+                    candidate_news_index_batch.append(candidate_news_index_batch[cnt%data_size])
+                    candidate_news_val_batch.append(candidate_news_val_batch[cnt%data_size])
+                    candidate_news_entity_index_batch.append(candidate_news_entity_index_batch[cnt%data_size])
+                    cnt += 1
```

Comment:
This function has some logics that are not tested in our unit-tests. It would be nice to have unit tests for this iterator. The only test we have is:

```
iterator = DKNTextIterator(hparams, tf.Graph())
assert iterator is not None
```

Reply:
I think we can do this on another PR